jetty-server/src/main/java/org/eclipse/jetty/server/Dispatcher.java

```diff
@@ -195,8 +196,24 @@ protected void forward(ServletRequest request, ServletResponse response, Dispatc
                 baseRequest.setContextPath(_contextHandler.getContextPath());
                 baseRequest.setServletPath(null);
                 baseRequest.setPathInfo(_pathInContext);
-                if (_uri.getQuery()!=null || old_uri.getQuery()!=null)
-                    baseRequest.mergeQueryParameters(old_uri.getQuery(),_uri.getQuery(), true);
+    
+                if (_uri.getQuery() != null || old_uri.getQuery() != null)
+                {
+                    try
+                    {
+                        baseRequest.mergeQueryParameters(old_uri.getQuery(), _uri.getQuery(), true);
+                    }
+                    catch (BadMessageException e)
+                    {
+                        // Only throw BME if not in Error Dispatch Mode
+                        // This allows application ErrorPageErrorHandler to handle BME messages
+                        Boolean inErrorDispatch = (Boolean) request.getAttribute(__ERROR_DISPATCH);
```

Comment:
I don't think you need to check the attribute (which is just re-entrancy protection).  Rather you can just check the dispatch type passed into the method.